app/models/mixins/retirement_mixin.rb

```diff
@@ -254,8 +254,12 @@ def valid_zone?
 
   def system_context_requester
     if try(:evm_owner_id).present?
-      if User.find_by(:id => evm_owner_id).present?
+      user = User.find_by(:id => evm_owner_id)
+      if user.present? && user.current_group.present?
         return evm_owner
+      elsif user.present? && user.current_group.blank?
+        $log.info("#{name} has evm_owner present but user has no group so defaulting to admin.")
+        return User.super_admin
       else
         $log.info("#{name} has evm_owner_id present but no user with that id found so defaulting to admin.")
         return User.super_admin
```

Comment:
```
  user = User.find_by(:id => evm_owner_id)
  return evm_owner_id if user.present? && user.current_group.present?
    
  if user.present?
    msg = "#{name} has evm_owner present"
    if user.current_group.blank?
      msg += " but user has no group so defaulting to admin."
    else
      msg += " but no user with that id found so defaulting to admin."
    end
  else
    msg = "System context defaulting to admin user because owner of #{name} not set."
  end
  $log.info(msg)
  User.super_admin
end
```

Reply:
Okay, but every time we do this we make it more difficult to do backports.